charmstore.go

```diff
@@ -359,11 +359,21 @@ func (s *CharmStore) GetResource(id *charm.URL, revision int, name string) (resu
 	}, nil
 }
 
-// Publish tells the charmstore to mark the given charm as published with the given resource revisions.
-func (s *CharmStore) Publish(id *charm.URL, resources map[string]int) (*charm.URL, error) {
+// Publish tells the charmstore to mark the given charm as published with the
+// given resource revisions to the given channels. If no channels are specified,
+// the stable channel will be assumed.
+func (s *CharmStore) Publish(id *charm.URL, channels []string, resources map[string]int) (*charm.URL, error) {
+	chans := make([]params.Channel, len(channels))
+	if len(channels) == 0 {
+		chans = []params.Channel{params.StableChannel}
+	}
+	for i, ch := range channels {
+		chans[i] = params.Channel(ch)
+	}
 	val := &params.PublishRequest{
 		Published: true,
```

Comment:
Delete this - it's ignored by the current API (and note TODO comment on the field in Publish).
You'll need to update the dependencies to use the latest charmstore version (which you should
do anyway so you're testing against a version that actually implements the functionality
we're trying to test).